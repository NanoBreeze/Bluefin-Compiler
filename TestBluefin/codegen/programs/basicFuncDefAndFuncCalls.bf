/*
Verifies basic function definitions, especially return type and param types
TODO: handle string and user-defined struct type
*/

bool hi() { }

void hey(bool a, float b, int c) {
}

float floatOnly(float a) {}

void intOnly(int a) {}

void boolOnly(bool a) {}

int bar() {
   	hi();
	floatOnly(5.0);
	intOnly(5);
	boolOnly(true);
	boolOnly(false);
}