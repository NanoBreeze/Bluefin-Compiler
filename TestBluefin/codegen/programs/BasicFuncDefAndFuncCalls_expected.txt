; ModuleID = 'BasicFuncDefAndFuncCalls.bf'
source_filename = "BasicFuncDefAndFuncCalls.bf"

define i1 @hi() {
entry:
}

define void @hey(i1 %a, float %b, i32 %c) {
entry:
}

define float @floatOnly(float %a) {
entry:
}

define void @intOnly(i32 %a) {
entry:
}

define void @boolOnly(i1 %a) {
entry:
}

define i32 @bar() {
entry:
  %calltmp = call i1 @hi()
  %calltmp1 = call float @floatOnly(float 5.000000e+00)
  call void @intOnly(i32 5)
  call void @boolOnly(i1 true)
  call void @boolOnly(i1 false)
}
